deposit/src/main/java/edu/unc/lib/deposit/work/AbstractDepositJob.java

```diff
@@ -370,7 +370,10 @@ protected void interruptJobIfStopped() throws JobInterruptedException {
         }
 
         DepositState state = getDepositStatusFactory().getState(depositId);
-        if (!DepositState.running.equals(state)) {
+
+        // Only throw error if state is changed and it's not a finished deposit waiting on cleanup job
+        if (!DepositState.running.equals(state) &&
+                (!DepositState.finished.equals(state) && !jobName.contains("CleanupDepositJob"))) {
```

Comment:
I'm not totally sure if the check of the job name is necessary, but interrupting the job if somehow it gets set to finished and isn't a cleanup job is probably fine since that would be an error at this time.
My only suggestion is that instead of doing a contains check on the job name, you can do `this instanceof CleanupDepositJob` instead, or maybe it needs to be `getClass()`, I'm not sure, but either would be good and a bit safer than a string check.